presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -163,21 +163,23 @@ private PlanBuilder handleSubqueries(PlanBuilder builder, Expression expression,
                 .collect(toImmutableSet());
     }
 
-    private PlanBuilder appendInPredicateApplyNodes(PlanBuilder subPlan, Set<InPredicate> inPredicates, boolean correlationAllowed)
+    private PlanBuilder appendInPredicateApplyNodes(PlanBuilder subPlan, Set<InPredicate> inPredicates, boolean correlationAllowed, Node node)
     {
         for (InPredicate inPredicate : inPredicates) {
-            subPlan = appendInPredicateApplyNode(subPlan, inPredicate, correlationAllowed);
+            subPlan = appendInPredicateApplyNode(subPlan, inPredicate, correlationAllowed, node);
         }
         return subPlan;
     }
 
-    private PlanBuilder appendInPredicateApplyNode(PlanBuilder subPlan, InPredicate inPredicate, boolean correlationAllowed)
+    private PlanBuilder appendInPredicateApplyNode(PlanBuilder subPlan, InPredicate inPredicate, boolean correlationAllowed, Node node)
     {
         if (subPlan.canTranslate(inPredicate)) {
             // given subquery is already appended
             return subPlan;
         }
 
+        subPlan = handleSubqueries(subPlan, inPredicate.getValue(), node);
```

Comment:
Is it not just beautiful?  Single line of code and all the subqueries in this expression are handled. The author of this code should be just proud of himself :) lol